nni/contrib/compression/utils/evaluator.py

```diff
@@ -723,17 +734,28 @@ def unbind_model(self):
         else:
             _logger.warning('Did not bind any model, no need to unbind model.')
 
-    def patch_loss(self, patch: Callable[[Tensor], Tensor]):
-        old_criterion = self._criterion
-
-        def patched_criterion(*args, **kwargs):
-            loss = old_criterion(*args, **kwargs)
-            return patch(loss)
-
-        self._criterion = patched_criterion
+    def patch_loss(self, patch: Callable[[Tensor, Any], Tensor]):
+        old_training_step = self._training_step
+
+        def patched_training_step(*args, **kwargs):
```

Comment:
I'm not sure whether this works, because Pytorch-lightning sometimes use inspect.parameters to inspect the arguments of `training_step`. If the signature has `batch_idx`, it will give you `batch_idx`. If the signature doesn't, it won't.

Reply:
could you give a link for this logic, I didn't find that